src/GovUk.Education.ExploreEducationStatistics.Data.Services.Tests/TableBuilderServiceTests.cs

```diff
@@ -259,6 +268,115 @@ await using (var statisticsDbContext = StatisticsDbUtils.InMemoryStatisticsDbCon
+                    EndCode = TimeIdentifier.AcademicYear
+                }
+            };
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await using (var statisticsDbContext = StatisticsDbUtils.InMemoryStatisticsDbContext(contextId))
+            {
+                await statisticsDbContext.AddAsync(releaseSubject);
+                await statisticsDbContext.SaveChangesAsync();
+            }
+
+            await using (var statisticsDbContext = StatisticsDbUtils.InMemoryStatisticsDbContext(contextId))
+            {
+                var filterItemRepository = new Mock<IFilterItemRepository>(MockBehavior.Strict);
+
+                filterItemRepository
+                    .Setup(s => s.CountFilterItemsByFilter(
+                        query.Filters))
+                    .ReturnsAsync(new Dictionary<Guid, int>
+                    {
+                        {
+                            // For the purpose of calculating the potential table size,
+                            // treat all the Filter Items as belonging to the same Filter
+                            Guid.NewGuid(), query.Filters.Count()
+                        }
+                    });
+
+                var subjectRepository = new Mock<ISubjectRepository>(MockBehavior.Strict);
+
+                subjectRepository
+                    .Setup(s => s.GetPublicationIdForSubject(query.SubjectId))
+                    .ReturnsAsync(publicationId);
+
+                subjectRepository
+                    .Setup(s => s.IsSubjectForLatestPublishedRelease(query.SubjectId))
+                    .ReturnsAsync(true);
+
+                var releaseRepository = new Mock<IReleaseRepository>(MockBehavior.Strict);
+
+                releaseRepository
+                    .Setup(s => s.GetLatestPublishedRelease(publicationId))
+                    .Returns(release);
+
+                var options = Options.Create(new TableBuilderOptions
+                {
+                    // 2 Filter items (from 1 Filter), 1 Location, and 2 Time periods provide 4 different combinations,
+                    // assuming that all the data is provided. For 2 Indicators this would be 8 table cells rendered.
+                    // Configure a maximum table size limit lower than 8.
+                    MaxTableCellsAllowed = 7
+                });
+
+                var service = BuildTableBuilderService(
+                    statisticsDbContext,
+                    filterItemRepository: filterItemRepository.Object,
+                    subjectRepository: subjectRepository.Object,
+                    releaseRepository: releaseRepository.Object,
+                    options: options
+                );
+
+                var result = await service.Query(query);
+
+                result.AssertBadRequest(ValidationErrorMessages.QueryExceedsMaxAllowableTableSize);
+
+                MockUtils.VerifyAllMocks(filterItemRepository, subjectRepository, releaseRepository);
```

Comment:
I'd put this immediately after `var result = await service.Query(query);`

Reply:
Thanks I've done this now and changed all the other tests in this class to use the same pattern.